File: src/main/java/de/twometer/amongus/gui/SettingsPage.java
package de.twometer.amongus.gui;

import de.twometer.amongus.core.AmongUs;
import de.twometer.amongus.util.UserSettings;

public class SettingsPage extends BasePage {

    public SettingsPage() {
        super("Settings.html");
    }

    @Override
    public void onDomReady() {
        super.onDomReady();
        context.call("set", AmongUs.get().getUserSettings());
    }

    public void apply(String settingsJson) {
        var settings = gson.fromJson(settingsJson, UserSettings.class);
        AmongUs.get().getUserSettings().copyFromJs(settings);
        AmongUs.get().getUserSettings().save();
        AmongUs.get().reloadConfig();
        goBack();
    }

    public void back() {
        goBack();
    }

}

File: src/main/kotlin/de/twometer/amogus/gui/SettingsPage.kt
package de.twometer.amogus.gui

import de.twometer.amogus.client.AmongUsClient
import de.twometer.amogus.client.GameConfig

class SettingsPage : BasePage("Settings.html") {

    override fun onLoaded() {
        call("set", AmongUsClient.gameConfig)
    }

    fun apply(settingsJson: String) {
        val settings = settingsJson.parseJson(GameConfig::class.java)
        AmongUsClient.gameConfig.set(settings)
        AmongUsClient.saveConfig()
        AmongUsClient.applyConfig()
        close()
    }

}